testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_delta_sync.py

```diff
@@ -94,9 +99,24 @@ def test_delta_sync_replication(params_from_base_test_setup, num_of_docs, replic
         cbl_db_docs = db.getDocuments(cbl_db, doc_ids)
         for doc_id, doc_body in cbl_db_docs.items():
             for i in range(number_of_updates):
-                doc_body["new_field_1"] = random_string(length=30)
-                doc_body["new_field_2"] = random_string(length=80)
-                db.updateDocument(database=cbl_db, data=doc_body, doc_id=doc_id, attachments_name=file_attachment)
+                if file_attachment:
+                    mutable_dictionary = dictionary.toMutableDictionary(doc_body)
+                    dictionary.setString(mutable_dictionary, "new_field_1", random_string(length=30))
+                    dictionary.setString(mutable_dictionary, "new_field_2", random_string(length=80))
+
+                    if liteserv_platform == "android":
+                        blob_value = blob.create("image/jpeg", blob.createImageContent("/assets/golden_gate_large.jpg"))
+                    elif liteserv_platform == "xamarin-android":
+                        image_content = blob.createImageContent("golden_gate_large.jpg")
+                        blob_value = blob.create("image/jpeg", stream=image_content)
+                    elif liteserv_platform == "ios":
+                        image_content = blob.createImageContent("Files/golden_gate_large.jpg")
+                        blob_value = blob.create("image/jpeg", content=image_content)
+                    else:
+                        image_content = blob.createImageContent("Files/golden_gate_large.jpg")
+                        blob_value = blob.create("image/jpeg", stream=image_content)
+                    dictionary.setBlob(mutable_dictionary, "_attachments", blob_value)
+                db.updateDocument(database=cbl_db, data=doc_body, doc_id=doc_id)
```

Comment:
I see that you created a new mutable_dictionary if there is new attachment update, however, you didn't pass it through. You are still using doc_body, which has same value as before update or I'm missing something?

Reply:
Good question. When we setBlob, it effects the doc_body directly . I had same question and I tested manually too to make sure doc_body after updating doc has blob and new fields.